astropy/utils/xml/tests/test_iterparse.py

```diff
@@ -0,0 +1,130 @@
+
+# SYSTEM
+try:
+    # Python 2.x
+    from StringIO import StringIO as BytesIO
+except ImportError:
+    # Python 3.x
+    from io import BytesIO
+import zlib
+
+# The C-based XML parser for VOTables previously used fixed-sized
+# buffers (allocated at __init__() time).  This test will
+# only pass with the patch that allows dynamic a realloc() of
+# the queue.  This addresses the bugs:
+#
+# - "RuntimeError: XML queue overflow"
+#   https://github.com/astropy/astropy/issues/5824
+#   (Kudos to Stefan Becker---ARI/ZAH Heidelberg)
+#
+# - "iterparse.c: add queue_realloc() + move 'buffersize / 2' logic there"
+#   https://github.com/astropy/astropy/issues/5869
+#
+# This test code can emulate a combination of network buffering and
+# gzip decompression---with different request sizes, it can be used to
+# demonstrate both under-reading and over-reading.
+# 
+# Using the 512-tag VOTABLE XML sample input, and various combinations
+# of minimum/maximum fetch sizes, the following situations can be
+# generated:
+#
+# maximum_fetch =  1 (ValueError, no element found) still within gzip headers
+# maximum_fetch = 80 (ValueError, unclosed token) short read
+# maximum_fetch =217 passes, because decompressed_length > requested
+#                            && <512 tags in a single parse
+# maximum_fetch =218 (RuntimeError, XML queue overflow)
+#
+# The test provided here covers the over-reading identified in #5824
+# (equivalent to the 217).
+
+# Firstly, assemble a minimal VOTABLE header, table contents and footer.
+# This is done in textual form, as the aim is to only test the parser, not
+# the outputter!
+HEADER = """<?xml version="1.0" encoding="UTF-8"?>
+<VOTABLE>
+ <RESOURCE type="results">
+  <TABLE>
+   <FIELD ID="foo" name="foo" datatype="int" arraysize="1"/>
+    <DATA>
+     <TABLEDATA>
+"""
+
+ROW = """<TR><TD>0</TD></TR>
+"""
+
+FOOTER = """
+    </TABLEDATA>
+   </DATA>
+  </TABLE>
+ </RESOURCE>
+</VOTABLE>
+"""
+
+# minimum passable buffer size => 1024
+# 1024 / 2 => 512 tags for overflow
+# 512 - 7 tags in header, - 5 tags in footer = 500 tags required for overflow
+# 500 / 4 tags (<tr><td></td></tr>) per row == 125 rows required for overflow
+VOTABLE_XML = HEADER + 125*ROW + FOOTER
+
+# UngzipFileWrapper() behaves wraps an existing file-like Object,
+# decompressing the content content before returning it.
+# This therefore emulated the behaviour of the Python 'requests'
+# library when transparently decompressing Gzip HTTP responses.
+#
+# The critical behaviour is that---because of the
+# decompression---read() can return considerably more
+# bytes than were requested!  (But, can also return less).
+#
+# inspiration: http://stackoverflow.com/questions/4013843/how-to-wrap-file-object-read-and-write-operation-which-are-readonly
+class UngzipFileWrapper(object):
+    def __init__(self, fd, **kwargs):
+        self._file = fd
+        self._z = zlib.decompressobj(16 + zlib.MAX_WBITS)
+    def read(self, requested_length):
```

Comment:
Needs a space before `def` here. I am sure this is one of the reasons the PEP8 test is failing.